src/tools/analyzer/module_request.py

```diff
@@ -0,0 +1,186 @@
+
+    def execute(self, source, options):
+        if options['list']:
+            if options['merge']:
+                print("--merge has no effect with --list")
+            self.list_requests(source, options['pam'])
+        elif options['cid']:
+            cid = options['cid'][0]
+            if not cid:
+                print("sss_analyze request: error: argument"
+                      " --cid: expected 1 argument")
+            else:
+                self.track_request(source, options)
+        else:
+            self.parser.print_help()
+
+    def print_formatted(self, line):
+        # exclude backtrace logs
+        if line.startswith('   *  '):
+            return
+        # files source includes newline
+        fields = line.split("[")
+        cr_field = fields[2].split(":")[1]
+        cr = cr_field[5:]
+        if "refreshed" in line:
+            return
+        # CR Plugin name
+        if re.search("cache_req_send", line):
+            plugin = line.split('\'')[1]
+            print("   - " + plugin)
+        # CR Input name
+        elif re.search("cache_req_process_input", line):
+            name = line.rsplit('[')[-1]
+            if cr not in self.done:
+                print("       - " + name[:-2])
+                self.done = cr
+        # CR Input id
+        elif re.search("cache_req_search_send", line):
+            id = line.rsplit()[-1]
+            if cr not in self.done:
+                print("       - " + id)
+                self.done = cr
+        # CID and client process name
+        else:
+            ts = line.split(")")[0]
+            fields = line.split("[")
+            cid = fields[3][5:-1]
+            cmd = fields[4][4:-1]
+            print(ts + ": " + "CID #" + cid + ": " + cmd)
+
+    def list_requests(self, source, pam):
+        component = source.Component.NSS
+        resp = "nss"
+        patterns = ['\[cmd']
+        patterns.append("(cache_req_send|cache_req_process_input|"
+                        "cache_req_search_send)")
+        consume = True
+        if pam:
+            component = source.Component.PAM
+            resp = "pam"
+
+        print(f"******** Listing {resp} client requests ********")
+        source.set_component(component)
+        self.done = ""
+        # For each CID
+        for match in self.matched_line(source, patterns):
+            self.print_formatted(match)
+
+    def track_request(self, source, options):
+        merge = options['merge']
+        client_id = options['cid'][0]
+        resp_results = False
+        be_results = False
+        component = source.Component.NSS
+        resp = "nss"
+        pattern = [f'REQ_TRACE.*\[CID #{client_id}\\]']
+        pattern.append(f"\[CID #{client_id}\\].*connected")
+
+        if options['pam']:
+            component = source.Component.PAM
+            resp = "pam"
+            pam_data_regex = f'pam_print_data.*\[CID #{client_id}\]'
+
+        print(f"******** Checking {resp} responder for Client ID"
```

Comment:
this looks like something that should go to a logger instead of stdout

Reply:
Sorry I don't understand what is needed here, can you elaborate?

Reply:
Python has built-in logging framework: https://docs.python.org/3/howto/logging.html

It might be nice to use that instead of print?